lib/luahid.c

```diff
@@ -50,6 +52,42 @@ static const lunatik_class_t luahid_class = {
 	.sleep = true,
 };
 
+static const struct hid_device_id *luahid_setidtable(lua_State *L, int idx)
+{
+	if (lua_getfield(L, idx, "id_table") != LUA_TTABLE) /* id_table */
+		goto err;
+
+	size_t len = luaL_len(L, -1);
+	if (len == 0)
+		goto err;
+
+	struct hid_device_id *user_table = lunatik_checkalloc(L, sizeof(struct hid_device_id) * (len + 1));
+
+	for (size_t i = 0; i < len; i++) {
+		if (lua_geti(L, -1, i + 1) != LUA_TTABLE) { /* i-th entry of the table */
+			lunatik_free(user_table);
+			luaL_error(L, "id_table entry #%zu is not a table", i + 1);
```

Comment:
I'm not sure about raising error in here.. I think it's make harder to read the register function.. perhaps it's better to return NULL and raise an error on register() instead.. I would like also to hear @sheharyaar's inputs on this..